### PR TITLE
Indicate in docs that stack monitoring uses a different index

### LINKED ISSUE #23774: [doc] Clarification that xpack metricbeat modules are not using metricbeat index
**Describe the enhancement:**
To add a note inside the metricbeat module saying this one is using monitoring index and a link to [elasticsearch explanation](https://www.elastic.co/guide/en/elasticsearch/reference/current/config-monitoring-indices.html).

https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-logstash.html
https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-elasticsearch.html
https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-beat.html

This is to avoid confusion loading metricbeat template when using logstash.
https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-template.html#load-template-manually

### PR DESCRIPTION
**Pull request by dedemorton**
Closes elastic/beats#23774

### CODE REVIEW

**Comment by andresrc** [comment]
IIRC, the use of a different index depends on whether we are using the xpack flavor or not /cc @sayden @ycombinator

**Comment by ycombinator** [comment]
I see that this PR enhances the documentation for the [`beat`](https://beats_23916.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-module-beat.html), [`kibana`](https://beats_23916.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-module-kibana.html), and [`logstash`](https://beats_23916.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-module-logstash.html) modules. Is there some reason it doesn't cover the [`elasticsearch`](https://beats_23916.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-module-elasticsearch.html) module as well?

File: metricbeat/docs/modules/beat.asciidoc [hunk 1]
```diff
@@ -20,6 +20,9 @@ UI in {kib}. To enable this usage, set `xpack.enabled: true` and remove any `met
 from the module's configuration. Alternatively, run `metricbeat modules disable beat` and
 `metricbeat modules enable beat-xpack`.
 
+NOTE: This module sends metrics to the monitoring index instead of the default
```
Comment (ycombinator) [inline metricbeat/docs/modules/beat.asciidoc:23]: I understand from looking at the [preview](https://beats_23916.docs-preview.app.elstc.co/guide/en/beats/metricbeat/master/metricbeat-module-beat.html#_usage_for_stack_monitoring) that this note renders under the "Usage for Stack Monitoring" section, which talks about the `xpack.enabled: true` flag. For extra clarity, however, could we qualify this note by saying something like, "When this module is used for Stack Monitoring, it sends metrics...".

File: metricbeat/docs/modules/beat.asciidoc [hunk 2]
```diff
@@ -20,6 +20,9 @@ UI in {kib}. To enable this usage, set `xpack.enabled: true` and remove any `met
...
+index typically used by {metricbeat}. For more details about the monitoring
+index, see {ref}/config-monitoring-indices.html.
```
Comment (bmorelli25) [inline metricbeat/docs/modules/beat.asciidoc:25]: Maybe add link text for all of these? Or is there a use-case for omitting it? This is how it looks right now:

Screen Shot 2021-02-09 at 7 09 06 AM

Comment (dedemorton) [reply metricbeat/docs/modules/beat.asciidoc:25]: That was a mistake. I will fix it. :-)

**Comment by dedemorton** [comment]
>I see that this PR enhances the documentation for the beat, kibana, and logstash modules. Is there some reason it doesn't cover the elasticsearch module as well?

I must have made the rookie mistake of updating the generated file. :-P 

Should be fixed now, along with some other minor changes, like using the correct attribute to resolve the name of the UI.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Pull request merged**